lib/commons/text/is-icon-ligature.js

```diff
@@ -0,0 +1,210 @@
+		else if (font.numLigatures === 0) {
+			return false;
+		}
+
+		// we could theoretically get into an odd middle ground scenario in which
+		// the font family is being used as normal text sometimes and as icons
+		// other times. in these cases we would need to always check the text
+		// to know if it's an icon or not
+	}
+	font.occurances++;
+
+	// 30px was chosen to account for common ligatures in normal fonts
+	// such as fi, ff, ffi. If a ligature would add a single column of
+	// pixels to a 30x30 grid, it would not meet the statistical significance
+	// threshold of 15% (30x30 = 900; 30/900 = 3.333%). this also allows for
+	// more than 1 column differences (60/900 = 6.666%) and things like
+	// extending the top of the f in the fi ligature.
+	let fontSize = 30;
+	let fontStyle = `${fontSize}px ${fontFamily}`;
+
+	// set the size of the canvas to the width of the first letter
+	context.font = fontStyle;
+	const firstChar = nodeValue.charAt(0);
+	let width = context.measureText(firstChar).width;
+
+	// ensure font meets the 30px width requirement (30px font-size doesn't
+	// necessarily mean its 30px wide when drawn)
+	if (width < 30) {
+		const diff = 30 / width;
+		width *= diff;
+		fontSize *= diff;
+		fontStyle = `${fontSize}px ${fontFamily}`;
+	}
+	canvas.width = width;
+	canvas.height = fontSize;
+
+	// changing the dimensions of a canvas resets all properties (include font)
+	// and clears it
+	context.font = fontStyle;
+	context.textAlign = 'left';
+	context.textBaseline = 'top';
+	context.fillText(firstChar, 0, 0);
+	const compareData = new Uint32Array(
+		context.getImageData(0, 0, width, fontSize).data.buffer
+	);
+
+	// if the font doesn't even have character data for a single char then
+	// it has to be an icon ligature (e.g. Material Icon)
+	if (!compareData.some(pixel => pixel)) {
+		font.numLigatures++;
+		return true;
+	}
+
+	context.clearRect(0, 0, width, fontSize);
+	context.fillText(nodeValue, 0, 0);
+	const compareWith = new Uint32Array(
+		context.getImageData(0, 0, width, fontSize).data.buffer
+	);
+
+	// calculate the number of differences between the first letter and the
+	// entire string, ignoring color differences
+	const differences = compareData.reduce((diff, pixel, i) => {
+		if (pixel === 0 && compareWith[i] === 0) {
+			return diff;
+		}
+		if (pixel !== 0 && compareWith[i] !== 0) {
+			return diff;
+		}
+		return ++diff;
+	}, 0);
+
+	// calculate the difference between the width of each character and the
+	// combined with of all characters
+	const expectedWidth = nodeValue.split('').reduce((width, char) => {
+		return width + context.measureText(char).width;
+	}, 0);
+	const actualWidth = context.measureText(nodeValue).width;
+
+	const pixelDifference = differences / compareData.length;
+	const sizeDifference = 1 - actualWidth / expectedWidth;
+
+	if (
+		pixelDifference >= differenceThreshold &&
+		sizeDifference >= differenceThreshold
```

Comment:
right now it's fine but later we might find we need to split pixel and size diff thresholds.